pkg/commands/git.go

```diff
@@ -470,7 +478,15 @@ func (c *GitCommand) GetHeadCommitMessage() (string, error) {
 func (c *GitCommand) AmendHead() (*exec.Cmd, error) {
 	command := "git commit --amend --no-edit --allow-empty"
 	if c.usingGpg() {
-		return c.OSCommand.ExecutableFromString(fmt.Sprintf("%s %s %s", c.OSCommand.Platform.shell, c.OSCommand.Platform.shellArg, command)), nil
+		quotedCommand := ""
```

Comment:
given there's quite a bit of duplicated code here now, could you factor out the two chunks of code into a single function? I.e. everything inside the if condition. We could call it something like 'runCommandInShell' and put it in `os.go`. What are your thoughts?

Reply:
Sure, I can factor it out into `RunCommandInShell`, below `ExecutableFromString`.

Reply:
Instead of `RunCommandInShell`, how about `ShellCommandFromString`, or something similar, since we aren't actually running the command.

Reply:
yep that works for me